src/Uno.UI/UI/Xaml/Controls/ContentPresenter/ContentPresenter.cs

```diff
@@ -852,6 +858,7 @@ public void UpdateContentTemplateRoot()
 			{
 				// No template and Content is a View, set it directly as root
 				ContentTemplateRoot = contentView as View;
+				IsUsingDefaultTemplate = false;
```

Comment:
Should this be reset if there's no content ?

Reply:
Adjusted to fallback to `false` in all cases except when implicit text block is present